### PR TITLE
Milestone 4 comments and likes

### PR DESCRIPTION
**Pull request by MarioDena**
Hello we added the functionality needed for the project to be approved

Comments to post linked to users
Likes on posts unique per user

We also added unlike feature

We hope this feature can be added to our development branch. Cheers!

### CODE REVIEW

File: app/views/comments/update.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<h1>Comments Update</h1>
+
+<%= form_for @comment do |f| %>
+    <%= f.label :content %>
+    <%= f.text_area :content, class: 'form-control' %>
+    <!-- <%= f.hidden_field :post_id, value: @post.id %> -->
```
Comment (ebukaume) [inline app/views/comments/update.html.erb]: Leaving commented code in your commit is a bad practice. Please remove.

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -7,10 +7,10 @@
...
    <li> <%= gravatar_for u, size: 50 %></li>
     <li><%= link_to 'View user', user_path(u) %></li>
-    <li>Nickname: <%= u.nickname %></li>
-    <li>Location: <%= u.location %></li>
-    <li>Gender: <%= u.gender %></li>
-   <li> Age: <%= u.age %></li>
+    <li>Nickname: <%= u.nickname || "Nickname not set" %></li>
+    <li>Location: <%= u.location || "We don't know where you are" %></li>
+    <li>Gender: <%= u.gender || "We dont know what you are" %></li>
+   <li> Age: <%= u.age || "We don't know how old you are" %></li>
```
Comment (ebukaume) [inline app/views/users/index.html.erb]: [OPTIONAL]
Consider abstracting these into a helper method. 

For exmaple: 
```
def nickname(user)
  user.nickname || 'Nickname not set'
end
```

File: app/views/comments/update.html.erb (hunk 1 above)
Comment (mikenath223) [reply app/views/comments/update.html.erb]: Sure will do that. Thanks for picking that out.

File: app/views/users/index.html.erb (hunk 2 above)
Comment (mikenath223) [reply app/views/users/index.html.erb]: Alright working on that now.

**Comment by mikenath223** [comment]
Oh sorry about that. But this is quite funny because my partner and i can both like posts. Try creating a post and then click the like button. Once more sorry about that. 
We will also implement the recommended tests. Thanks

**Comment by ebukaume** [comment]
> Oh sorry about that. But this is quite funny because my partner and i can both like posts. Try creating a post and then click the like button. Once more sorry about that.
> We will also implement the recommended tests. Thanks

That's exactly what I did. I had to `clone the repo, bundle, setup db, sign up, create post and try to like the post`. I tried it twice.

**Comment by mikenath223** [comment]
Hey perhaps you can try liking directly from the show post path `/posts/2`
Kindly let us know how it goes.

**Comment by mikenath223** [comment]
Hi thanks for the review. But from the guide we were requested to add unit(model) and integrations test. Just as we did. Do well to confirm again.

**Comment by mikenath223** [comment]
Yea same problem we encountered initially. It has to do with postgres database

**Comment by MarioDena** [comment]
Added the test everything should be working

**Comment by samgaco** [comment]
Hi team :high_brightness: 

You can move on to the next milestone :+1: 

----------------------

* ###### STATUS OF THE MILESTONE: APPROVED :heavy_check_mark:

**Comment by mikenath223** [comment]
Great to hear. Really appreciate all the feedback. Thanks.

**Pull request merged**